File: daily/2025/1432_Max_Difference_You_Can_Get_From_Changing_an_Integer.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int maxDiff(int num) {
        string s = to_string(num);
        string t = s;
        int n = s.length();
        int idx = 0;
        while(idx < n && s[idx] == '9') idx++;
        char c = s[idx];
        if (idx < n) replace(s.begin(), s.end(), c, '9');
        if (t[0] != '1'){
            c = t[0];
            replace(t.begin(), t.end(), c, '1');
        }
        else{
            idx = 1;
            while(idx < n && (t[idx] == '1' || t[idx] == '0')) idx++;
            c = t[idx];
            replace(t.begin(), t.end(), c, '0');
        }
        // cout << s << " " << t << endl;
        return stoi(s) - stoi(t);
    }
};

int main(){
    int num = 555;
    // int num = 9;
    // int num = 1919;
    Solution S;
    cout << S.maxDiff(num) << endl;
    return 0;
}

File: daily_python/1432_Max_Difference_You_Can_Get_From_Changing_an_Integer.py
import os
import sys
from typing import List

class Solution:
    def maxDiff(self, num: int) -> int:
        s = str(num)
        t = str(num)
        for d in s:
            if d != '9':
                s = s.replace(d, '9')
                break
        for i, d in enumerate(t):
            if i == 0 and d != '1':
                t = t.replace(d, '1')
                break
            elif d != '0' and d != '1':
                t = t.replace(d, '0')
                break
        return int(s) - int(t)
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            num = int(lines[0].split('=')[1].strip()[:-1])
            dataset.append(num)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for num in dataset:
        results.append(solution.maxDiff(num))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')